sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -406,6 +406,13 @@ private void initializeGatewayConfigurationReader() {
         // https://msdata.visualstudio.com/CosmosDB/_workitems/edit/332589
     }
 
+    private void updateGatewayProxy() {
```

Comment:
nit: the naming is weird. maybe `configureGatewayProxy` ?

Reply:
configureGatewayProxy  means our proxy object is not ready to use, which is not the case, here we are updating it with some caches and boolean (useMultipleWriteLocations)